### PR TITLE
test: add registry unit test

### PR DESCRIPTION
**Pull request by binbin-li**
### What?
Add unit tests for the `registry` package.
Resolves https://github.com/notaryproject/notation-go/issues/69

image

Signed-off-by: Binbin Li <libinbin@microsoft.com>

### CODE REVIEW

File: registry/repository.go [hunk 1]
```diff
@@ -30,6 +30,9 @@ type SignatureManifest struct {
 	Annotations map[string]string
 }
 
+// var for unit testing
+var fetchAll = content.FetchAll
```
Comment (chloeyin) [inline registry/repository.go:34]: Can we use a mock service instead? E.g. https://github.com/googlemaps/google-maps-services-go/blob/391036138cfed529e87aaad49922e96da4a393da/directions_test.go#L41

File: registry/repository_test.go [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+
+type mockRemoteClient struct {
+}
+
+func (c mockRemoteClient) Do(req *http.Request) (*http.Response, error) {
+	switch req.URL.Path {
+	case "/v2/test/manifests/" + validDigest, "/v2/test/blobs/" + validDigestWithAlgo:
+		return &http.Response{
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader([]byte(msg))),
```
Comment (chloeyin) [inline registry/repository_test.go:103]: nit: `ioutil ` will be depreciated(https://go.dev/doc/go1.16), we can use package `io` instead.

File: registry/repository.go (hunk 1 above)
Comment (binbin-li) [reply registry/repository.go]: updated.

File: registry/repository_test.go (hunk 2 above)
Comment (binbin-li) [reply registry/repository_test.go]: good call, addressed.

**Comment by chloeyin** [comment]
lgtm

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by dtzar** [comment]
One more approval from any of the notation-code-reviewers and we're good to merge since we're over the week.

**Review by chloeyin — APPROVED** [review]

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by chloeyin — APPROVED** [review]

**Pull request merged**